### ISSUE TITLE
Support ManagedPolicies

### ISSUE THREAD

**Issue by eytan-avisror** [keikoproj/instance-manager#54]
instance-manager should support managedPolicies in InstanceGroup custom resource.
This should add managed policies to the role that is created for the instance group.

**Comment by eytan-avisror** [comment]
How should we implement this:

- We need to extend the InstanceGroup API to take `.spec.eks-cf.configuration.managedPolicies` which is a list of managed policies which should be attached to the cloudformation template.

- We should assume that the provided policies already exist in the account we are operating in.

- In the current instance-manager config map that is in `config/crd/bases/instance-manager-configmap.yaml` there is a config map that holds the cloudformation template being used. we should modify the section that creates the IAM role and add templating for the additional managed policies (without modifying the ones that are already there and are needed)

```
      NodeInstanceRole:
        Type: AWS::IAM::Role
        Properties:
         ...
         ...
          ManagedPolicyArns:
            - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
            - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
            - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
            < Add go templating here to iteratively add the requested additional managed policies>
```

- Make sure to also take your changes to `docs/04_instance-manager.yaml` which has a copy of the configmap

- It will be nice to check whether each policy hasPrefix `arn:aws:iam::aws:policy/` and try to add it if not, however not sure current structure has a good place to add this in, we can do this with this PR or a future PR.

Essentially a user will provide input via the CR like so:
```
apiVersion: instancemgr.keikoproj.io/v1alpha1
kind: InstanceGroup
metadata:
  name: hello-world
spec:
  provisioner: eks-cf
  strategy:
    ...
  eks-cf:
    ...
    configuration:
      ...
      managedPolicies:
        - arn:aws:iam::aws:policy/My-Managed-Policy-1
        - arn:aws:iam::aws:policy/My-Managed-Policy-2
        - arn:aws:iam::aws:policy/My-Managed-Policy-3
```

Which would change the cloudformation template to:

```
...
ManagedPolicyArns:
            - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
            - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
            - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
            - arn:aws:iam::aws:policy/My-Managed-Policy-1
            - arn:aws:iam::aws:policy/My-Managed-Policy-2
            - arn:aws:iam::aws:policy/My-Managed-Policy-3
...
```

**Comment by eytan-avisror** [comment]
closed by #55
